flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/deduplicate/MiniBatchDeduplicateKeepLastRowFunctionTest.java

```diff
@@ -113,4 +113,42 @@ public void testWithGenerateRetraction() throws Exception {
 		testHarness.close();
 		assertor.assertOutputEqualsSorted("output wrong.", expectedOutput, testHarness.getOutput());
 	}
+
+	@Test
+	public void tesKeepLastRowWithStateTtlAndGenerateRetraction() throws Exception {
+		MiniBatchDeduplicateKeepLastRowFunction func = createFunction(true, minTime.toMilliseconds());
+		OneInputStreamOperatorTestHarness<BaseRow, BaseRow> testHarness = createTestHarness(func);
+		testHarness.setup();
+		testHarness.open();
+
+		testHarness.processElement(record("book", 1L, 10));
+		testHarness.processElement(record("book", 2L, 11));
+		// output is empty because bundle not trigger yet.
+		Assert.assertTrue(testHarness.getOutput().isEmpty());
+		testHarness.processElement(record("book", 1L, 13));
+
+		testHarness.setStateTtlProcessingTime(30);
+		//Incremental cleanup is an eventual clean up, more state access guarantee more expired state cleaned
+		for (long i = 3; i < 30; i++){
```

Comment:
nit: `) {`

Reply:
just curious, if no more other state access, old state will be valid when it is accessed?

Reply:
sorry, I have updated it.
Yes, Incremental cleanup is eventual cleanup, needs more state access, it keeps a global iterator and might not have reached and cleaned some expired state.

Reply:
have you tested it？
I'm asking because in state backend, we can easily check the ttl when we get the state, and even it's not cleaned physically.

Reply:
Yeah, I have tested it. When remove these code, the test failed. You can see [FLINK-10473](https://issues.apache.org/jira/browse/FLINK-10473?focusedCommentId=17068492&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17068492) for more explain.